packages/esm-reports-app/src/components/report-data-table.component.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import {
+  DataTable,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '@carbon/react';
+import classNames from 'classnames';
+import styles from './reports.scss';
+
+interface ReportDataTableProps {
+  data: {
+    dataSets: Array<{
+      metadata: {
+        columns: Array<{
+          name: string;
+          label: string;
+          datatype: string;
+        }>;
+      };
+      rows: Array<Record<string, any>>;
+    }>;
+  };
+}
+
+const ReportDataTable: React.FC<ReportDataTableProps> = ({ data }) => {
+  if (!data?.dataSets?.[0]) {
+    return null;
+  }
+
+  const { metadata, rows } = data.dataSets[0];
+  const columns = metadata.columns;
+
+  const tableHeaders = columns.map((column) => ({
+    key: column.name,
+    header: column.label,
+  }));
+
+  const tableRows = rows.map((row, index) => ({
+    id: `row-${index}`,
+    ...row,
+  }));
+
+  return (
+    <DataTable rows={tableRows} headers={tableHeaders} isSortable>
+      {({ rows, headers }) => (
+        <TableContainer>
+          <Table>
+            <TableHead>
+              <TableRow>
+                {headers.map((header) => (
+                  <TableHeader key={header.key}>{header.header}</TableHeader>
+                ))}
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {rows.map((row) => (
+                <TableRow key={row.id} className={styles.tableRow}>
+                  {row.cells.map((cell) => (
+                    <TableCell
+                      key={cell.id}
+                      className={classNames({
+                        [styles.rowCellEven]: row.index % 2 === 0,
+                        [styles.rowCellOdd]: row.index % 2 !== 0,
+                      })}
+                    >
+                      {formatCellValue(cell.value)}
+                    </TableCell>
+                  ))}
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
+        </TableContainer>
+      )}
+    </DataTable>
+  );
+};
+
+const formatCellValue = (value: any): string => {
+  if (value === null || value === undefined) {
+    return '';
+  }
+
+  if (Array.isArray(value)) {
+    // Handle date arrays [year, month, day, hour, minute, second]
+    if (value.length >= 3 && value.length <= 6) {
+      const [year, month, day, hour = 0, minute = 0, second = 0] = value;
+      const date = new Date(year, month - 1, day, hour, minute, second);
+      return date.toLocaleString();
```

Comment:
We have a `formatDateTime()` function in the framework I think it would be preferable to use here.